inductiva/fluids/scenarios/wind_tunnel/post_processing.py

```diff
@@ -0,0 +1,86 @@
+"""Visualization processing of WindTunnel scenario.
+
+This class implements various visualization capabilities for
+the WindTunnel scenario. Namely:
+    - Pressure over object; 
+    - Cutting plane;
+    - StreamLines.
+
+Currently, we only support the OpenFOAM simulator.
+"""
+from functools import singledispatchmethod
+
+import os
+import pyvista as pv
+
+from inductiva.simulation import Simulator
+from inductiva.fluids.simulators import OpenFOAM
+from inductiva.types import Path
+from inductiva.utils.visualization import Object
+
+
+class WindTunnelSimulationOutput:
+    """Post process WindTunnel simulation outputs."""
+
+    def __init__(self,
+                 sim_output_path: Path,
+                 time_step: int,
+                 simulator: Simulator = OpenFOAM()):
+        """Initializes an `OpenFOAMSimulationOutput` object.
+
+        Args:
+            simulator: Simulator object.
+            sim_output_path: Path to simulation output files.
+            time_step: Time step where we read the data.
+        
+        Attributes:
+            sim_output_dir: path to the simulation directory
+            object_path: path to the object insert in the WindTunnel
+            post_processing_path: path to the post-processing data.
+        """
+
+        self.sim_output_dir = sim_output_path
+        self.object_path = self.get_object
+        self.object_data = self.get_object_data(simulator, time_step)
+        self.pressure_field = self.get_pressure_field(simulator)
+
+    @singledispatchmethod
```

Comment:
I am confused about why you need this...

Reply:
The goal is to be able to overload these functions for different simulators. Check out FluidBlock that already does this for SPlisHSPlasH and DualSPHysics. Before this was done, we had multiple if...else...

Reply:
You have the answer in your question :)